### PR TITLE
Fixes typos

### PR DESCRIPTION
**Pull request by deining**
This PR fixes  typos I spotted in various files. It also converts two links from `http` to `https` protocol.

### CODE REVIEW

File: src/eval/operation.rs [hunk 1]
```diff
@@ -594,7 +594,7 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
                         let mut shared_env = Environment::new();
                         let f_as_var = f.body.closurize(&mut self.cache, &mut env, f.env);
 
-                        // Array elements are closurized to preserve lazyness of data structures. It
+                        // Array elements are closurized to preserve laziness of data structures. It
```
Comment (yannham) [inline src/eval/operation.rs:597]: Ah, I've always thought lazyness was a valid US spelling. But it seems it's actually not the case.

File: src/types.rs [hunk 2]
```diff
@@ -1050,7 +1050,7 @@ mod test {
     /// Take a string representation of a type, parse it, and assert that formatting it gives the
     /// same string as the original argument.
     ///
-    /// Note that their are infintely many string representations of the same type since, for
+    /// Note that their are infinitely many string representations of the same type since, for
```
Comment (yannham) [inline src/types.rs:1053]:
```suggestion
    /// Note that there are infinitely many string representations of the same type since, for
```

**Review by yannham — APPROVED** [review]
Thanks for the clean up! As a side note, If you've read the in-code documentation yourself, be ware that while individual items (attached to methods, types, etc.) are mostly up-to-date, the top-level documentation of some modules is a bit outdated.

File: src/eval/operation.rs (hunk 1 above)
Comment (deining) [reply src/eval/operation.rs:597]: So no change on my side here ...

**Pull request merged**